nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -595,41 +655,172 @@ sub multiLineList {
   # services.openssh.enable = true;
 
   # Open ports in the firewall.
+  #
   # networking.firewall.allowedTCPPorts = [ ... ];
   # networking.firewall.allowedUDPPorts = [ ... ];
-  # Or disable the firewall altogether.
+  #
+  # WARNING! WARNING! WARNING! INSECURE!
+  #
+  # To disable the firewall altogether use the following.
+  #
+  # Use only inside trusted networks behind another firewall or else
+  # bad people across the galaxy will be able to access any ports you
+  # leave open.
+  #
   # networking.firewall.enable = false;
 
   # Enable CUPS to print documents.
   # services.printing.enable = true;
 
-  # Enable sound.
+  # Enable ALSA. This option will add ALSA tools to your
+  # systemPackages and make ALSA settings such as volume persistent
+  # across reboots. Not enabling this doesn't actually disable sound
+  # output via ALSA because ALSA is the default sound API of the Linux
+  # kernel. Also see `sound.mediaKeys.enable` in configuration.nix(5).
+  #
   # sound.enable = true;
+
+  # Enable sound output via PulseAudio. This option will enable
+  # PulseAudio daemon and configure `asound.conf` to route the
+  # `default` ALSA pcm and ctl to PulseAudio daemon.
+  #
   # hardware.pulseaudio.enable = true;
+  #
+  # Note that you don't need to enable PulseAudio daemon to get either
+  # of sound playback or software sound mixing. It will out of the box
+  # over ALSA. If you do want to run PulseAudio daemon, however, you
+  # may safely disable `sound.enable`.
+
+  # Also note that you can emulate PulseAudio API over ALSA for apps
+  # that refuse to work over ALSA themselves by running them under
+  # apulse. This way you can run one daemon less which benefits your
+  # security (some people claim that PulseAudio codebase lacks in
+  # quality to be connected to untrusted data sources like the browser
+  # by default).
+  #
+  # environment.systemPackages = [ pkgs.apulse ];
 
   # Enable the X11 windowing system.
+  #
   # services.xserver.enable = true;
   # services.xserver.layout = "us";
   # services.xserver.xkbOptions = "eurosign:e";
-
+  #
   # Enable touchpad support.
   # services.xserver.libinput.enable = true;
 
-  # Enable the KDE Desktop Environment.
-  # services.xserver.displayManager.sddm.enable = true;
-  # services.xserver.desktopManager.plasma5.enable = true;
+  # Enable graphical acceleration for NVIDIA GPUs (requires unfree
+  # packages, see above).
```

Comment:
:-1: this information is too likely to get out of date, and really can't cover all the important cases. This should lead to external documentation.